### PR TITLE
M3-1604 close support ticket

### PR DESCRIPTION
**Pull request by Jskobos**
Add functionality to close a support ticket (one that has been
marked by the API as "closable").

## To Test
`closeTicket.json` mock will allow you to walk through the flow. Alternatively, you can open a ticket through the Manager, then set it to AutoClose in Admin.

## Notes
* Added a success Notice since otherwise it seemed a little abrupt. Can remove if it's unnecessary.
* Removed scrollToBottom when loading the support ticket detail view. This has been on the chopping block for a while, and conflicted with scrollToTop on ticket close. Again, feedback welcome.

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworking.tsx [hunk 1]
```diff
@@ -154,10 +154,10 @@ class LinodeNetworking extends React.Component<CombinedProps, State> {
     this.setState({ IPRequestError: undefined });
     return getLinodeIPs(this.props.linodeID)
       .then(ips => this.setState({ linodeIPs: ips, initialLoading: false }))
-      .catch((errors) => {
-        const IPRequestError = pathOr('There was an error retrieving your network information.',
-          ['reason'], errors[0]);
-        this.setState({ IPRequestError, initialLoading: false });
+      .catch((errorResponse) => {
```
Comment (patthiel) [inline src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworking.tsx]: Was this supposed to be changed? Seems like this was already done in `src/features/Support/SupportTicketDetail/CloseTicketLink.tsx`

Comment (Jskobos) [reply src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworking.tsx]: Yeah,  I tried the old pattern here in CloseTicketLink and it didn't work as I was expecting, so I changed this one to match the pattern that works.

File: src/features/Support/SupportTicketDetail/CloseTicketLink.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    dialogOpen: false,
+    isClosingTicket: false,
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = true;
```
Comment (martinmckenna) [inline src/features/Support/SupportTicketDetail/CloseTicketLink.tsx]: false

Comment (Jskobos) [reply src/features/Support/SupportTicketDetail/CloseTicketLink.tsx]: oops. fixed

**Pull request merged**